wavesexchange_topic/src/lib.rs

```diff
@@ -116,12 +118,25 @@ pub struct LeasingBalance {
     pub address: String,
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct ExchangePair {
+    pub amount_asset: String,
+    pub price_asset: String,
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct ExchangePairs {
+    pub pairs: Vec<ExchangePair>,
```

Comment:
Making this field public allows to create invalid instances, so the drawback of the previous implementation is not solved. Better to make it private + add public constructor with all necessary checks (at least that the vector is not empty).